tests/test_kernel_execution.py

```diff
@@ -245,3 +245,19 @@ def MoveWest(particle, fieldset, time, dt):
     pset.execute(pset.Kernel(MoveEast), endtime=1., dt=1.)
     pset.execute(pset.Kernel(MoveWest), endtime=2., dt=1.)
     assert np.allclose([p.lon for p in pset], 0.3, rtol=1e-5)  # should be 0.5 + 0.1 - 0.3 = 0.3
+
+
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+def test_errorcode_repeat(fieldset, mode):
+    def simpleKernel(particle, fieldset, time, dt):
```

Comment:
I don't really understand what's happening in this kernel. Can you add a short explanation of its rationale?